src/main/scala/com/wavesplatform/account/Address.scala

```diff
@@ -17,44 +18,59 @@ sealed trait Address extends AddressOrAlias {
 
 //noinspection ScalaDeprecation
 object Address extends ScorexLogging {
-  val Prefix = "address:"
-
+  val Prefix               = "address:"
   val AddressVersion: Byte = 1
   val ChecksumLength       = 4
   val HashLength           = 20
   val AddressLength        = 1 + 1 + HashLength + ChecksumLength
   val AddressStringLength  = base58Length(AddressLength)
 
-  def fromPublicKey(publicKey: Array[Byte], chainId: Byte = scheme.chainId): Address = {
-    val withoutChecksum = ByteBuffer
-      .allocate(1 + 1 + HashLength)
-      .put(AddressVersion)
-      .put(chainId)
-      .put(crypto.secureHash(publicKey), 0, HashLength)
-      .array()
+  private[this] val publicKeyBytesCache: Cache[ByteStr, Address] = CacheBuilder.newBuilder()
+    .weakKeys()
```

Comment:
> <p><b>Warning:</b> when this method is used, the resulting cache will use identity (<code> == </code>) comparison to determine equality of keys.

This should probably be `Cache[PublicKey, Address]`: weak keys will work fine for interned instances.